internal/app/proxyd/internal/frontend/frontend_test.go

```diff
@@ -2,13 +2,164 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-package frontend_test
+package frontend
 
-import "testing"
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
 
-func TestEmpty(t *testing.T) {
-	// added for accurate coverage estimation
-	//
-	// please remove it once any unit-test is added
-	// for this package
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/suite"
+	"github.com/talos-systems/talos/internal/app/proxyd/internal/backend"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+type ProxydSuite struct {
+	suite.Suite
+}
+
+func TestFrontendSuite(t *testing.T) {
+	// Hide all our state transition messages
+	log.SetOutput(ioutil.Discard)
+	suite.Run(t, new(ProxydSuite))
+}
+
+func (suite *ProxydSuite) TestWatch() {
+	_, cancel := context.WithCancel(context.Background())
+	r, err := NewReverseProxy([]string{"127.0.0.1"}, cancel)
+	suite.Assert().NoError(err)
+	defer r.Shutdown()
+
+	// Generate a simple pod
+	p := genPod()
+
+	// Create our fake k8s client
+	client := fake.NewSimpleClientset()
+	go r.Watch(client)
+
+	output := make(chan string)
+	go func() {
+		var be map[string]*backend.Backend
+		for {
+			be = r.Backends()
+			if _, ok := be[string(p.UID)]; ok {
+				output <- be[string(p.UID)].UID
+			}
+		}
+	}()
+
+	// Verify we have a bootstrap backend
+	_, err = client.CoreV1().Pods("kube-system").Create(p)
+	suite.Assert().NoError(err)
+
+	timeout := time.NewTicker(time.Second * 5)
+	select {
+	case <-timeout.C:
+		suite.T().Error("failed to get updated backend")
+	case uid := <-output:
+		suite.Equal(string(p.UID), uid)
+	}
+}
+
+func (suite *ProxydSuite) TestAddBackendFunc() {
```

Comment:
nit: Convention is to name the test funcs the same as the func they test, only with the prefix `Test`.
`s/TestAddBackendFunc/TestAddFunc/`

Reply:
leftover from copy paste